src/js/controllers/index.js

```diff
@@ -138,21 +138,28 @@ angular.module('copayApp.controllers').controller('indexController', function($r
       self.initGlidera();
 
       self.setCustomBWSFlag();
-      if (fc.isPrivKeyExternal()) {
-        self.needsBackup = false;
+      self.isBackupNeeded(fc, function(needsBackup) {
+        self.needsBackup = needsBackup;
         self.openWallet();
-      } else {
-        storageService.getBackupFlag(self.walletId, function(err, val) {
-          if (!fc.credentials.mnemonic)
-            self.needsBackup = false;
-          else
-            self.needsBackup = self.network == 'testnet' ? false : !val;
-          self.openWallet();
-        });
-      }
+      });
     });
   };
 
+  self.isBackupNeeded = function(fc, cb) {
+    if (fc.isPrivKeyExternal()) return cb(false);
+
```

Comment:
from here it should be something like:

if (!fc.cretentials.mnemonic) return cb(false);
if (... testnet) return cb(false)

... getBackupFlag